base/namedtuple.jl

```diff
@@ -103,6 +103,8 @@ firstindex(t::NamedTuple) = 1
 lastindex(t::NamedTuple) = nfields(t)
 getindex(t::NamedTuple, i::Int) = getfield(t, i)
 getindex(t::NamedTuple, i::Symbol) = getfield(t, i)
+getindex(t::NamedTuple, v::AbstractVector{Symbol}) = [t[i] for i in v]
```

Comment:
This maps all `v::AbstractVector` to `Vector`.

I wonder if we can make this more generic, as in something like `map(Fix1(getindex, t), v)`? For example, that definition would mean if `v` were a `SVector` than the output could also be a `SVector`.

Reply:
Replied below. This seems like a fine implementation if we want to move towards the semantics in your Julep.